Feature/add separate parameter steps to tutorial (#3)
chrisguida/cln-plugin-tutorial: .tutorial-rust/04-rpc-methods.md

l1-cli testmethod
```

> Now, lets use the `serde_json::value` parameter which allows us to pass arguments.

Reviewer: ```suggestion
Now, let's use the `serde_json::Value` parameter which allows us to pass arguments.
```

Create a new method `testmethod_argument` and register it in your main() function.
Change `_v` to `v` as it is no longer an unused parameter.

```rust
.rpcmethod("testmethod_argument", "This is a test", testmethod_argument)
```

```rust
async fn testmethod_argument(p: Plugin<()>, v: serde_json::Value) -> Result<serde_json::Value, Error> {
  Ok(json!(format!("Hello {}", v)))
}
```

Try it with an argument:

```
l1-cli testmethod_argument <your name>
```

The v argument is passed to our RPC method as json.

> Parsing the string and printing just the string is left as an exercise to the reader.

Reviewer: ```suggestion
Parsing and printing just the string (properly formatted, without escape characters) is left as an exercise to the reader.
```

> Bonus is to print an error if more than 1 argument is passed.

Reviewer: ```suggestion
As a bonus exercise, print an error if more than 1 argument is passed.
```
